### ISSUE TITLE
cluster_leiden with multilevel outputs

### ISSUE THREAD

**Issue by youngser** [igraph/rigraph#568]
Hi!
My understanding is that both Louvain and Leiden are iterative algorithm, and
`cluster_louvain` outputs multilevel labels  in `memberships`, while `cluster_leiden` doesn't.
I wonder if there is a way or plan to add this to `cluster_leiden` function?
Thanks,

**Comment by ntamas** [comment]
Summon @vtraag - I am not familiar with the internals of the Leiden algorithm so I don't know whether it's feasible.

**Comment by vtraag** [comment]
Thanks for your question!

In my opinion, the "intermediate" results (i.e. the different steps in the aggregation) in the  Louvain algorithm does not make much sense (but the implementation faithfully implements the original publication in that regard). The idea of the Louvain algorithm is to optimise a particular quality function, and the "intermediate" results are known to not optimise this particular quality function. For that reason, I think it does not make much sense.

In the Leiden algorithm, the concept of the "intermediate" results is even more complicated, as the aggregation is done in a different way, based on some refinement of the actual partition. In other words, I think it makes even less sense to present such "intermediate" results in the Leiden algorithm. If you are interested in more (or less) fine-grained clusters you can always simply adjust the resolution parameter as you see fit.

So, in short, there are no plans to add this to the `cluster_leiden` function. Hope that answers your question! I'll close the issue, but if you have any other questions about this, feel free to continue the conversation.
